WordPressAuthenticatorTests/Authenticator/WordPressAuthenticatorTests.swift

```diff
@@ -55,4 +65,190 @@ class WordPressAuthenticatorTests: XCTestCase {
 
         XCTAssert(email != retrievedEmail, "Saved loginFields should be deleted after calling deleteLoginInfoForTokenAuth.")
     }
+
+//MARK: WordPressAuthenticator Notification Tests
+    func testDispatchesSupportPushNotificationReceived() {
+        let authenticator = MockWordpressAuthenticatorProvider.getWordpressAuthenticator()
+        let _ = expectation(forNotification: .wordpressSupportNotificationReceived, object: nil, handler: nil)
+
+        authenticator.supportPushNotificationReceived()
+
+        waitForExpectations(timeout: timeInterval, handler: nil)
+    }
+
+    func testDispatchesSupportPushNotificationCleared() {
+        let authenticator = MockWordpressAuthenticatorProvider.getWordpressAuthenticator()
+        let _ = expectation(forNotification: .wordpressSupportNotificationCleared, object: nil, handler: nil)
+
+        authenticator.supportPushNotificationCleared()
+
+        waitForExpectations(timeout: timeInterval, handler: nil)
+    }
+
+//MARK: View Tests
+    func testWordpressAuthIsAuthenticationViewController() {
+        let loginViewcontroller = LoginViewController()
+        let nuxViewController = NUXViewController()
+        let nuxTableViewController = NUXTableViewController()
+        let basicViewController = UIViewController()
+
+
+        XCTAssertTrue(WordPressAuthenticator.isAuthenticationViewController(loginViewcontroller))
+        XCTAssertTrue(WordPressAuthenticator.isAuthenticationViewController(nuxViewController))
+        XCTAssertTrue(WordPressAuthenticator.isAuthenticationViewController(nuxTableViewController))
+        XCTAssertFalse(WordPressAuthenticator.isAuthenticationViewController(basicViewController))
+    }
+
+    func testShowLoginFromPresenterReturnsLoginInitialVC() {
+        let presenterSpy = ModalViewControllerPresentingSpy()
+        let expectation = XCTNSPredicateExpectation(predicate: NSPredicate(block: { (_, _) -> Bool in
+            return presenterSpy.presentedVC != nil
+        }), object: .none)
+
+        WordPressAuthenticator.showLoginFromPresenter(presenterSpy, animated: true)
+        wait(for: [expectation], timeout: 3)
+
+        XCTAssertTrue(presenterSpy.presentedVC is LoginNavigationController)
+    }
+
+    func testShowLoginForJustWPComPresentsCorrectVC() {
+        let presenterSpy = ModalViewControllerPresentingSpy()
+        let expectation = XCTNSPredicateExpectation(predicate: NSPredicate(block: { (_, _) -> Bool in
+            return presenterSpy.presentedVC != nil
+        }), object: .none)
+
+        WordPressAuthenticator.showLoginForJustWPCom(from: presenterSpy)
+        wait(for: [expectation], timeout: 3)
```

Comment:
Should this use `timeInterval` (or `timeout` as per my earlier suggestion), too?